### PR TITLE
add  annotation to kubeconfig to identify contexts/clusters created by minikube

### PR DESCRIPTION
**Pull request by medyagh**
After this PR the following extension blocks to the context and cluster section of  ~/.kube/config
```
apiVersion: v1
clusters:
- cluster:
    certificate-authority: /Users/medya/.minikube/ca.crt
    extensions:
    - extension:
        last-update: Thu, 14 Jan 2021 15:41:04 PST
        provider: minikube.sigs.k8s.io
        version: v1.16.0
      name: cluster_info
    server: https://127.0.0.1:55000
  name: minikube
contexts:
- context:
    cluster: minikube
    extensions:
    - extension:
        last-update: Thu, 14 Jan 2021 15:41:04 PST
        provider: minikube.sigs.k8s.io
        version: v1.16.0
      name: context_info
    namespace: default
    user: minikube
  name: minikube
current-context: minikube
kind: Config
preferences: {}
users:
- name: minikube
  user:
    client-certificate: /Users/medya/.minikube/profiles/minikube/client.crt
    client-key: /Users/medya/.minikube/profiles/minikube/client.key


```

closes https://github.com/kubernetes/minikube/issues/9970

### CODE REVIEW

**Comment by sharifelgamal** [comment]
do we want to use hyphens instead of underscores (`created-by` and `last-update` instead of `created_ by` and `last_update`) to follow the convention set by kubeconfig?

File: pkg/minikube/kubeconfig/settings.go [hunk 1]
```diff
@@ -83,6 +85,14 @@ func PopulateFromSettings(cfg *Settings, apiCfg *api.Config) error {
 	} else {
 		cluster.CertificateAuthority = cfg.CertificateAuthority
 	}
+
+	lastUpdate := time.Now().String()
```
Comment (briandealwis) [inline pkg/minikube/kubeconfig/settings.go:89]: Can this use `.Format(time.RFC3339)`?

Comment (sharifelgamal) [reply pkg/minikube/kubeconfig/settings.go:89]: also it doesn't seem like assigning a variable here serves any real purpose. i don't think lastUpdate gets reused anywhere, right?

Comment (briandealwis) [reply pkg/minikube/kubeconfig/settings.go:89]: It may help humans who edit their `~/.kube/config` to determine how fresh this section might be?

**Comment by briandealwis** [comment]
I get your dislike of using `cluster-type` since they're Kubernetes clusters.  But my concern about `created-by` is that it can refer to a tool like terraform.  I wonder if there's a different term that we could settle on like `provider`.  Basically within Skaffold I want to determine which helper command Skaffold should use to introspect on the cluster to find out more about its setup.

File: pkg/minikube/kubeconfig/settings.go [hunk 2]
```diff
@@ -149,3 +160,36 @@ func Update(kcs *Settings) error {
 	}
 	return nil
 }
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
```
Comment (priyawadhwa) [inline pkg/minikube/kubeconfig/settings.go:164]: I think it would be useful to move these functions into internal_extension.go for organization

**Comment by medyagh** [comment]
> I get your dislike of using `cluster-type` since they're Kubernetes clusters. But my concern about `created-by` is that it can refer to a tool like terraform. I wonder if there's a different term that we could settle on like `provider`. Basically within Skaffold I want to determine which helper command Skaffold should use to introspect on the cluster to find out more about its setup.

I dont think it is a good idea to call it cluster-type when minikube creates the cluster, minikube clusters are not a "type" they are same clusters created by standard "kubedam" it would be misleading or confusing for normal users to see in their kubeconfig it call it "cluster_type"= minikube

File: pkg/minikube/kubeconfig/settings.go (hunk 2 above)
Comment (medyagh) [reply pkg/minikube/kubeconfig/settings.go:164]: good idea

File: pkg/minikube/kubeconfig/settings.go (hunk 1 above)
Comment (medyagh) [reply pkg/minikube/kubeconfig/settings.go:89]: good catch the var was initially used two places got lost in refactor.

Comment (medyagh) [reply pkg/minikube/kubeconfig/settings.go:89]: good call on formatting.

File: pkg/minikube/kubeconfig/settings.go (hunk 2 above)
Comment (medyagh) [reply pkg/minikube/kubeconfig/settings.go:164]: good idea

**Comment by medyagh** [comment]
> I get your dislike of using `cluster-type` since they're Kubernetes clusters. But my concern about `created-by` is that it can refer to a tool like terraform. I wonder if there's a different term that we could settle on like `provider`. Basically within Skaffold I want to determine which helper command Skaffold should use to introspect on the cluster to find out more about its setup.

"provider" sounds like a good suggestion

**Comment by medyagh** [comment]
> I get your dislike of using `cluster-type` since they're Kubernetes clusters. But my concern about `created-by` is that it can refer to a tool like terraform. I wonder if there's a different term that we could settle on like `provider`. Basically within Skaffold I want to determine which helper command Skaffold should use to introspect on the cluster to find out more about its setup.

"provider" sounds like a good suggestion

**Review by priyawadhwa — APPROVED** [review]

**Comment by medyagh** [comment]
> ad of underscores (`created-by` and `last-update` instead of `created_ by` and `last_update`) to follow the convention set by kubeconfig?

good idea

**Pull request merged**